gpbft/api.go

```diff
@@ -54,9 +54,11 @@ type MessageReceiver interface {
 
 // Interface which network participants must implement.
 type Receiver interface {
-	// Begins executing the protocol.
+	// Beings the execution of the instance number passed as an argument.
 	// The node will request the canonical chain to propose from the host.
-	Start() error
+	// This function cna be used to skip directly to an instance number,
+	// for instance, when a new valid finality certificate is received
+	SkipToInstance(uint64) error
```

Comment:
Two questions:

* Can you please remind me where the finality certificate gets fed in that justifies the jump? My first thought looking at this API was that it should take the justification for skipping ahead as an argument but I am conscious that I might be missing something.

* What happens if there is no justification for the jump? What errors should the user expect? We probably want to document the behaviour.

Reply:
> Can you please remind me where the finality certificate gets fed in that justifies the jump? My first thought looking at this API was that it should take the justification for skipping ahead as an argument but I am conscious that I might be missing something.

Initially, I thought so too, but gpbft participants have no knowledge of certificates, signatures, when the protocol is stuck or how to compute the corresponding power tables. Only the host knows all of this. What gpbft needs to expose is an interface that the host can use to skip to the instance once a certificate has been validated, and the new power tables have been stored in the host. I guess the next step is to wire this into the host and the certificate validation.

> What happens if there is no justification for the jump? What errors should the user expect? We probably want to document the behaviour.

Thiis is a good point. This function can't return an error anymore, same as `Start()`. With this new approach errors triggered will be handled once the instance has begun. I will push the corresponding update in my next commit.